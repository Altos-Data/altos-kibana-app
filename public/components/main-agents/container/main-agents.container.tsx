/*
 * Wazuh app - React component for main overview
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { useEffect, useState } from 'react';
import { compose } from 'redux';
import { withGlobalBreadcrumb } from '../../common/hocs/withGlobalBreadcrumb';
import { WzAgents } from '../../agents/index';
import AppState from '../../../react-services/app-state';
import { useHistory } from 'react-router-dom';
import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';

export const MainAgents = compose(withGlobalBreadcrumb([{ text: '' }, { text: 'Agents' }]))(
  () => {
     return (
       <WzAgents></WzAgents>
    );
  }
);