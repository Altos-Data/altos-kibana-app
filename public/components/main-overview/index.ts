import overviewCurrentSection from '../common/modules/overview-current-section';

export { MainOverview } from './container/main-overview.container';