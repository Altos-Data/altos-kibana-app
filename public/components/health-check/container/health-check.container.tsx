/*
 * Wazuh app - Health Check Component
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import {
  EuiButton,
  EuiCallOut,
  EuiDescriptionList,
  EuiLoadingSpinner,
  EuiSpacer,
} from '@elastic/eui';
import ErrorHandler from '../../../react-services/error-handler';
import React from 'react';
import { useState, useEffect } from 'react';
import { useAppConfig } from '../../../components/common/hooks/use-app-config';
import {
  checkPatternService,
  checkTemplateService,
  checkApiService,
  checkSetupService,
  checkFieldsService,
  checkKibanaSettings,
  checkKibanaSettingsTimeFilter,
} from '../services';
import { CheckResult } from '../components/check-result';
import AppState from '../../../react-services/app-state';
import { useAppDeps } from '../../../components/common/hooks/use-app-deps';
import { Fragment } from 'react';

const checks = {
  api: {    
    title: 'Check Wazuh API connection',
    validator: checkApiService,
    awaitFor: []
  },
  setup: {
    title: 'Check for Wazuh API version',
    validator: checkSetupService,
    awaitFor: ["api"],
  },
  pattern: {
    title: 'Check Elasticsearch index pattern',
    validator: checkPatternService,
    awaitFor: []
  },
  template: {
    title: 'Check Elasticsearch template',
    validator: checkTemplateService,
    awaitFor: ["pattern"],
  },
  fields: {
    title: 'Check index pattern fields',
    validator: checkFieldsService,
    awaitFor: ["pattern"],
  },
};

export function HealthCheck() {
  const [checkErrors, setCheckErrors] = useState<any[]>([]);
  const [checksReady, setChecksReady] = useState<{[key: string]: boolean}>({});
  const { core } = useAppDeps();
  const appConfig = useAppConfig();

  useEffect(() => {
    if (appConfig.isReady) {
      checkKibanaSettings(appConfig.data['checks.metaFields']);
      checkKibanaSettingsTimeFilter(appConfig.data['checks.timeFilter']);
      AppState.setPatternSelector(appConfig.data['ip.selector']);
    }
  }, [appConfig]);

  const handleErrors = (errors, parsed) => {
    const newErrors = parsed
      ? errors.map((error) =>
          ErrorHandler.handle(error, 'Health Check', { warning: false, silent: true })
        )
      : errors;
    setCheckErrors((prev) => [...prev, ...newErrors]);
  };

  const handleCheckReady = (checkId, isReady) => {    
    setChecksReady(prev =>  ({...prev, [checkId]: isReady}));
  }

  const renderChecks = () => {
    return Object.keys(checks).map((check, index) => {
      return (
        <CheckResult
          key={index}          
          name={check}
          title={checks[check].title}
          awaitFor={checks[check].awaitFor}
          check={appConfig.data[`checks.${check}`]}
          validationService={checks[check].validator}
          handleErrors={handleErrors}
          isLoading={appConfig.isLoading}
          handleCheckReady= {handleCheckReady}
          checksReady={checksReady}
        />
      );
    });
  };

  const renderErrors = () => {
    return checkErrors.map((error, index) => {
      return (
        <Fragment key={index}>
          <EuiCallOut
            title={error}
            color="danger"
            iconType="alert"
            style={{ textAlign: 'left' }}
          ></EuiCallOut>
          <EuiSpacer size="xs" />
        </Fragment>
      );
    });
  };

  const logo_url = core.http.basePath.prepend('/plugins/wazuh/assets/icon_blue.svg');
  return (
    <div className="health-check">
      <EuiLoadingSpinner className="health-check-loader" />
      <img src={logo_url} className="health-check-logo" alt=""></img>
      <div className="margin-top-30">
        <EuiDescriptionList textStyle="reverse" align="center" type="column">
          {renderChecks()}
        </EuiDescriptionList>
      </div>
      <EuiSpacer size="xxl" />
      {renderErrors()}
      <EuiSpacer size="xxl" />
      {!!checkErrors.length && (
        <EuiButton fill href="/settings">
          Go to App
        </EuiButton>
      )}
    </div>
  );
}
