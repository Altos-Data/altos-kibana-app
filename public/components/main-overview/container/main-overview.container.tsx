/*
 * Wazuh app - React component for main overview
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { useEffect, useState } from 'react';
import { compose } from 'redux';
import AppState from '../../../react-services/app-state';
import { withGlobalBreadcrumb } from '../../common/hocs/withGlobalBreadcrumb';
import { OverviewWelcome } from '../components/overview-welcome';
import { OverviewStats } from '../components/overview-stats';
import { useHistory } from 'react-router-dom';
import { useAgentsSummary } from '../../common/hooks/agents/use-agents-summary';
import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';

export const MainOverview = compose(withGlobalBreadcrumb([{ text: '' }, { text: 'Modules' }]))(
  () => {
    const [extensions, setExtensions] = useState({});
    const [agentsSummaryLoading, agentsSummary, agentsSummaryError] = useAgentsSummary();
    const history = useHistory();

    useEffect(() => {
      const currentApi = JSON.parse(AppState.getCurrentAPI() || '{}');
      if (!currentApi.id) {
        history.push('/health-check');
      }
      AppState.getExtensions(currentApi.id).then((_extensions) => {
        setExtensions(_extensions);
      });
    }, []);

    return (
      <EuiFlexGroup direction="column">
        <EuiFlexItem grow={false}>
          <OverviewStats summary={agentsSummary} />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <OverviewWelcome extensions={extensions} />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }
);
