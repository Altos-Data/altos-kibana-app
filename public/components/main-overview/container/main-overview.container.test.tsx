/*
 * Wazuh app - React component for main overview - Test
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { MainOverview } from './main-overview.container';

jest.mock('../../../react-services/app-state', () => ({
  getCurrentAPI: () => {
    id: 'default';
  },
  getExtensions: () => [],
}));

jest.mock('../../common/hocs/withGlobalBreadcrumb', () => {
  return {
    withGlobalBreadcrumb: () => {
      return (component) => {
        component.defaultProps = {
          ...component.defaultProps,
        };
        return component;
      };
    },
  };
});

jest.mock('../../common/hooks/agents/use-agents-summary', () => ({
  useAgentsSummary: () => [false, {}, undefined],
}));

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('../components/overview-welcome', () => ({
  OverviewWelcome: () => () => <></>,
}));

jest.mock('../components/overview-stats', () => ({
  OverviewStats: () => () => <></>,
}));

describe('Main overview container', () => {
  test('should render a Main overview screen', () => {
    const component = shallow(<MainOverview />);

    expect(component).toMatchSnapshot();
  });
});