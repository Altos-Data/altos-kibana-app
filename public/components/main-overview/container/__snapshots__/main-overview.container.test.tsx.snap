// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main overview container should render a Main overview screen 1`] = `
<EuiFlexGroup
  direction="column"
>
  <EuiFlexItem
    grow={false}
  >
    <OverviewStats
      summary={Object {}}
    />
  </EuiFlexItem>
  <EuiFlexItem
    grow={false}
  >
    <OverviewWelcome
      extensions={Object {}}
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;
